lib/pbench/server/indexer.py

```diff
@@ -3603,9 +3604,11 @@ def __init__(self, idxctx, tbarg, tmpdir, extracted_root):
                     'run.controller ("{}") does not match'
                     ' controller_dir ("{}")'.format(controller, self.controller_dir)
                 )
+            self.controller = controller
             name = self.mdconf.get("pbench", "name")
             if not name:
                 raise Exception("empty pbench.name")
+            self.name = name
```

Comment:
Perhaps add a comment that this is a convenience field since this `name` value is also available as `self.run_metadata["name"]?

Also, we might want to validate that this value is a prefix of the tar ball name while we are in here.

Reply:
> Also, we might want to validate that this value is a prefix of the tar ball name while we are in here.

Yes, we could probably be doing a lot more consistency checking here; although, in a way, I'm tempted to say that in the long run maybe relying on consistent metadata rather than file name isn't an entirely bad idea. If we have the metadata, why should we enforce arbitrary rules on the filename? (Aside from the fact that clearly in other places we still rely on such assumptions... is there really any value in that?)